astropy/table/table.py

```diff
@@ -2262,6 +2262,21 @@ def __len__(self):
             self._first_colname = next(iter(self.columns))
             return len(self.columns[self._first_colname])
 
+    def __or__(self, other):
+        try:
+            updated_table = self.copy()
+            updated_table.update(other, copy=True)
```

Comment:
For `dict` updates, no copies are made. Am happy to go with the default for `.update()` here (but would then remove `copy=True` here), since perhaps it is too risky otherwise. But thought I would mention it, since it is also possible the main reason for `copy=True` is that when tables were based on structured arrays, there really wasn't an option, and that we wanted to be backward-compatible. But this is a new operator, so we can decide differently.